### DISCUSSION TITLE
lightweight version of arduino-audio-tools...

### DISCUSSION THREAD

**Discussion by mkoenings** [pschatzmann/arduino-audio-tools#1163] [Q&A] [answered]
I was trying to do some simple stuff with an Arduino Nano:  reading from an rf24 radio  (rf24 library) and writing into a codec (VS1003). But oh boy, it was 1900 bytes variable memory short... The Atmel 328 has only 2kbyte available.  I tried to read the Data as WAV into a SD, same issue. 
Is there a way to "shrink" or reduce the library ( i guess it's the culprit...)  to match small simple tasks on small MCUs ? Thanks!

**Comment by pschatzmann** [comment, marked as answer]
I can not recommend the Arduino Nano for any serious audio. I suggest that you rather use a RP2040 if you con't need any communication or a STM32. The Nano just has not enough RAM, ROM and speed to be useful. In addition it does not have any DAC or ADC and does not support I2S...

You could deactivate the logging (by setting USE_AUDIO_LOGGING to false) (or decrease the LOG_PRINTF_BUFFER_SIZE) but then you will need to use a debugger to be able to figure out why something is not working. I would expect that you will also need to decrease other size defintions and there is still a high probability that you will run out of dynamically allocated ram...

**Reply by mkoenings** [reply]
you are right, the nano is not very powerful. I just got the ones with an nrf24 chip on it. I t would be awesome to use  it as an audio receiver, with the lowest specs possible. I got inspired by cheap audio transmitter/receiver modules that you can get on aliexpress. The use a nrf24, a Atmel mega 48 and a wolfson WM8738 audio chip to sport 44khz, 16 bit mono. I guess they are running optimized assembler code...
Thanks!
![IMG_20240107_124639](https://github.com/pschatzmann/arduino-audio-tools/assets/442877/afed7481-7ee6-424f-9ef1-99a449fc11b4)
![IMG_20240107_124630](https://github.com/pschatzmann/arduino-audio-tools/assets/442877/77ca0e8d-510f-4b5e-8a98-a330c580a8cf)

**Comment by pschatzmann** [comment]
It would be interesting to see how they pulled this one off: The WM8738 requires i2s which the nano does not support.

Did you check if you have enough memory by just using the rf24 and vs1003 libraries and just copy the data in the loop ?
The only thing that you will need to take care of, is to add a wav header before the transmission. In this situation just a hard coded prefix might be the best option.
